Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Pass.cpp

```diff
@@ -113,6 +114,15 @@ namespace AZ
             return desc;
         }
 
+        const int Pass::GetDeviceIndex() const
+        {
+            if (m_deviceIndex == AZ::RHI::MultiDevice::InvalidDeviceIndex && m_parent)
+            {
+                return m_parent->GetDeviceIndex();
+            }
+            return AZStd::max(m_deviceIndex, 0);
```

Comment:
Why do you have a max here? This should still return `InvalidDeviceIndex`?!

Reply:
Because the data to be uploaded to the device buffer are usually 'unordered_map<int, const void*>', and the index for device 0 there are default to `0`, so if someone want's to access the data there after using `Pass::GetDeviceIndex()`, we should always give a meaningful result.

But yes we can also do the max there after querying from `Pass::GetDeviceIndex()`, I can remove the max here. Just want to prevent writing this check everywhere needs a device index.

Reply:
Which data upload and which buffer? The device index of a pass should remain and also be reported as such, since this signals that the pass can be scheduled on any device, so yes, please remove the max.